### ISSUE TITLE
cf login -a https://api.local.pcfdev.io --skip-ssl-validation API endpoint: https://api.local.pcfdev.io   API endpoint:   https://api.local.pcfdev.io (API version: 2.54.0) Not logged in. Use 'cf login' to log in. FAILED Server error, status code: 503, error code: , message:

### ISSUE THREAD

**Issue by Veeresh-Patel** [vmware-archive/pcfdev#99]
Hi Guys,

PCFdev is up and running in my windows machine. But when i do a cf login from CMD terminal I get the above error message. When I ssh into the machine I am able to login to the endpoint successfully but am unable to do it directly from the windows cmd.

**Comment by sclevine** [comment]
Hi @Veeresh-Patel,

Can you provide your boot output? Have you tried PowerShell, or just `cmd.exe`?

Are you behind an HTTP/S proxy? Have you tried destroying PCF Dev, rebooting your host, and starting PCF Dev again?

Thanks,
Stephen

**Comment by amitshukla30** [comment]
Hi @sclevine ,

I received the similar kind of error on my Windows 10 machine with Oracle VM virtual box 5.1 using PCFDev 0.17. See the screenshot as below.

**_Command :_** cf login -a https://api.local.pcfdev.io --skip-ssl-validation

_Error :_ API endpoint: https://api.local.pcfdev.io
FAILED
Error performing request: Get https://api.local.pcfdev.io/v2/info: dial tcp: lookup api.local.pcfdev.io: getaddrinfow: No such host is known.
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.

![image](https://cloud.githubusercontent.com/assets/6778900/16823817/09ee8990-4967-11e6-97eb-c18a46a92c81.png)

Interestingly I am able to do PING on 192.168.11.11. I checked my VM Network Settings just to check if VM is able to access the internet. Do you have any suggestions on my network settings? Do i need to add more routing rules or change the network strategy. I had kept everything as default. Also I don't use any corporate proxy or something. 

![image](https://cloud.githubusercontent.com/assets/6778900/16823859/6580075c-4967-11e6-94f4-a5471aef027b.png)

I tried destroying the VM twice and started it up 2 times. Even did a complete reinstallation of Oracle Virtualbox. Still no luck. 

Regards
Amit Shukla

**Comment by sclevine** [comment]
@amitshukla30: The issue you describe is different. Your host appears unable to resolve the domain `api.local.pcfdev.io`. Can you `ping api.local.pcfdev.io`? The `*.local.pcfdev.io` record is handled by public DNS servers, so you may be using an oddly configured DNS server on the host that doesn't resolve wildcard alias records. If you change your host DNS to `8.8.8.8` and `8.8.4.4` (Google's public DNS servers), it will likely fix your issue.

(Note: Lookups to `*.local.pcfdev.io` domains are handled by a DNS server running inside the VM to improve performance. This is why the VM is not affected by your host's inability to resolve `api.local.pcfdev.io`. If you can't change the DNS servers on the host, you can follow [these instructions](https://docs.pivotal.io/pcf-dev/work-offline.html) to apply the same hack to the host.)

@Veeresh-Patel: I think your issue is proxy-related, but you didn't mention whether you're using an HTTP/HTTPS proxy to connect to the internet.

**Comment by Veeresh-Patel** [comment]
@sclevine : Yes, I am behind an HTTP/S proxy and I am using HTTP/HTTPS proxy to connect to the internet.
Even I tried from the bosh director VM(192.168.50.4), I am even not able to ping the api.local.pcfdev.io.
Even I tried by changing my host DNS to 8.8.8.8 and 8.8.4.4. No help.

**Comment by sclevine** [comment]
Hi @Veeresh-Patel,

PCF Dev does not have a BOSH director or use the IP address `192.168.50.4`. PCF Dev only runs on a single IP address, which is usually `192.168.11.11`.  Changing the host DNS will likely fix @amitshukla30's issue, but your issue is different.

I think your issue is that connections from the cf CLI plugin should not go through your HTTP/S proxy. Please add `.local.pcfdev.io` to your `no_proxy` environment variable in the shell you run `cf login` from, and attempt to connect again. You shouldn't need to destroy or restart the VM.

Thanks,
Stephen

**Comment by Veeresh-Patel** [comment]
Hi @sclevine,
Yes, It was a proxy issue. I have already added .local.pcfdev.io to my no_proxy environment variable in the shell. Actually my laptop was connected to corporate network and the proxy was different.
I used the proper proxy and it's working.

Thanks,
Veeresh

**Comment by Veeresh-Patel** [comment]
I am closing the issue.

**Comment by pratikkadam518** [comment]
i have same problem as mentioned above.What is solution

**Comment by akumar261089** [comment]
Add following entries in C:\Windows\System32\drivers\etc\hosts 
192.168.11.11 api.local.pcfdev.io
192.168.11.11 login.local.pcfdev.io

**Comment by pourgholi** [comment]
Had the same problem on ubuntu 18.04. Addinf the entries provided by @akumar261089 helped me:
`sudo vim /etc/hosts`

Add the entries to the file:

`192.168.11.11 api.local.pcfdev.io`
`192.168.11.11 login.local.pcfdev.io`

**Comment by ghost** [comment]
Hi ,
I had to prepare presentation for PCF and have the exact errors above irrelevant whether I run it on Centos or local MS-Windows toy it gives the same error:

[root@svmgmlo001:~/sandbox/spring-music]# cf login -a api.local.pcfdev.io --skip-ssl-validation
API endpoint: api.local.pcfdev.io
FAILED
Error performing request: Get https://api.local.pcfdev.io/v2/info: dial tcp 192.168.11.11:443: i/o timeout
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.

I can access the internet no worries and have updated the no_proxy var but no luck.

Then I started to do Public cloud deployment and that crashed too. Throwing out many errors.
As a result:
![web-error](https://user-images.githubusercontent.com/34564819/54272362-d9e1d880-4583-11e9-8ef4-0c623d5316f8.png)


![Bild1](https://user-images.githubusercontent.com/34564819/54270920-bd906c80-4580-11e9-9f2f-1f697390ba38.png)

So basically the 15 Minutes app deployment failed in both cases.
One would not expect to troubleshoot so much for a simple example.
This does not look good!
